HealthcareSystem/GraphicalEditor/Models/MapObjectRelated/Building.cs

```diff
@@ -23,5 +27,59 @@ public override void FormatObjectDescription(string description)
                 Description = MapObjectType.ObjectTypeFullName + " " + Id + " ima " + NumOfFloors + " spratova.";
             }
         }
+
+        public void AddBuildinLayersButtonsToCanvas(Canvas canvas)
+        {
+            List<Button> floorButtons = BuildingLayersButtons.CreateBuildingLayersButtons(NumOfFloors);
+            foreach (Button button in floorButtons)
+            {
+                button.Click += ShowChoosenFloor;
+                canvas.Children.Add(button);
+            }
+
+        }
+
+        public void LoadChoosenFloor(MapObject mapObject, int choosenFloor, Canvas canvas)
+        {
+            if (mapObject.MapObjectEntity.GetType() == typeof(Room) && ((Room)mapObject.MapObjectEntity).BuildingId == Id && ((Room)mapObject.MapObjectEntity).Floor == choosenFloor)
+            {
+                mapObject.AddToCanvas(canvas);
+            }
+        }
+
+        public void ClearOtherFloors(MapObject mapObject, int choosenFloor, Canvas canvas)
+        {
+            if (mapObject.MapObjectEntity.GetType() == typeof(Room) && ((Room)mapObject.MapObjectEntity).BuildingId == Id && ((Room)mapObject.MapObjectEntity).Floor != choosenFloor)
```

Comment:
This works for now, but having a static field in the MainWindow that this class accesses is generally a bad design choice. Maybe a redesign to a tree like structure for the entire map in the future might be a good idea - consider something like the Composite pattern? That would include moving around pretty much all the logic you have right now though, not just floors, but it might be worth it in the long term.

Reply:
@VasilijeBursac @kmilena98 @ogikuzma @djura023 consider this idea.